### DISCUSSION TITLE
All sites considered constant in SNP data set

### DISCUSSION THREAD

**Discussion by tylerbreech** [iqtree/iqtree2#214] [Q&A] [answered]
Greetings,
I am trying to run iqtree2 with a SNP dataset of 24 individuals. However, after creating a phylip format data set using vcf2phylip, making sure to have the SNP present in at least 20 individuals, iqtree is recognizing all the sites as constant sites, and thus the output file it gives to rerun is just a blank alignment. Is there a way to fix this? The code I am running is:

bin\iqtree2.exe -s 24NN.min20.phy -st DNA -m JC+ASC

and the output:
Reading alignment file 24NN.min20.phy ... Phylip format detected
Alignment most likely contains protein sequences
WARNING: Your specified sequence type is different from the detected one
Alignment has 24 sequences with 453895 columns, 317643 distinct patterns
0 parsimony-informative, 0 singleton sites, 453895 constant sites
              Gap/Ambiguity  Composition  p-value

Thanks for any help!

**Comment by roblanf** [comment]
Can you post your alignment (or a link to it)?

**Reply by tylerbreech** [reply]
Sure thing
https://drive.google.com/file/d/14cReDu2oyBfapKFxwF1zxl-xa56MM_Rh/view?usp=sharing

**Comment by roblanf** [comment, marked as answer]
@tylerbreech the issue seems to be that you have zero information in your alignment. If you use a later version of IQ-TREE (not sure which one you were using) it would give the following error:

```
ERROR: Invalid use of +ASC because of 453895 invariant sites in the alignment
```

So, if you run without `+ASC`, then you get the following tree:

```
(NN114296:0.0000001102,NN114297:0.0000001102,((NN115950:0.0000001102,SAMN32324430:0.0000001102):0.0000000000,((((((SAMN32324407:0.0000001102,SAMN32324426:0.0000001102):0.000000
2203,(((((SAMN32324408:0.0000001102,SAMN32324425:0.0000001102):0.0000000000,(SAMN32324409:0.0000001102,SAMN32324424:0.0000001102):0.0000000000):0.0000000122,(SAMN32324410:0.000
0001102,SAMN32324423:0.0000001102):0.0000002203):0.0000000138,(SAMN32324411:0.0000001102,SAMN32324422:0.0000001102):0.0000002203):0.0000000157,(SAMN32324412:0.0000001102,SAMN32
324421:0.0000001102):0.0000002203):0.0000000184):0.0000000220,(SAMN32324413:0.0000001102,SAMN32324420:0.0000001102):0.0000002203):0.0000000275,(SAMN32324414:0.0000001102,SAMN32
324419:0.0000001102):0.0000002203):0.0000000367,(SAMN32324415:0.0000001102,SAMN32324418:0.0000001102):0.0000002203):0.0000000551,(SAMN32324416:0.0000001102,SAMN32324417:0.00000
01102):0.0000002203):0.0000000000):0.0000000000);
```

In which you can see that ~all the branch lengths are at the lower bound. 

Examining your alignment makes it clear why. Here are the first few bases:

```
24 453895
NN114296       AYGGCGCYGYMYGTGRCRRNGRTCTACRRGCGGC
NN114297       NYGRYGCCGCCTGTGRCRGAGRTCTACRRGCGGC
NN115950       ACGGCGCYGYMYGTGACAGAGATCTACAAGCGGC
SAMN32324407   ACGGCGCCGCCTRYSRYAGAGRTCTACNNGCGGC
SAMN32324408   ACGGCGNCGCCTRYSRCRGAGAKCYACRRGCGGY
SAMN32324409   ACGGCGCCGCCTRYSRCRGAGAKCYACRRGCGGY
SAMN32324410   AYGGCGCCGCCYRYSRCAGRGRTCTACNRGCGGC
SAMN32324411   ACGGCGCYGCMTGTGRCRGAGAKSYACRAGCSGC
SAMN32324412   NNNNNNCNNNNNNNNNNAGAGATCTACAANNNNN
SAMN32324413   RYRRYRCNRNNNRYSRYAGAGATCTACAARYSRY
SAMN32324414   NNNNNNCNNNNNNNNNNAGAGATCTACAANNNNN
SAMN32324415   ACGGCGCCGCCYRTGRCNGAGATCTACAAGCGGC
SAMN32324416   ANGGCGCCGCCTGTGRCNRRGRTCTACRRGCGGC
SAMN32324417   ACGGCGCYGCMTRTGRCRGNGATCTACAAGCGGN
SAMN32324418   NNRRYRCYRYMYNYSRYAGAGATCTACRARYSRY
SAMN32324419   ACGGCGCNGYNYGTGRCARNGRKSYACRRGCSGC
SAMN32324420   ACGGCGCCGCCTGTGNCNGAGATCTACAAGCGGC
SAMN32324421   ACGGCGCYGCMTRTGRCRGAGATCTACAAGCGGC
SAMN32324422   ACGGCGCCGCCNNTGACNGAGATCTACAAGCGGC
SAMN32324423   RYRRYRCYRYMYRYSRYAGAGATCTACAARYSRY
SAMN32324424   RYRRYRCYRYMYRYSRYAGAGATCTACAARYSRY
SAMN32324425   RYGGCGCCGCCTRTGACAGAGATCTANRAGCGGC
SAMN32324426   ACGGCGCCGCCTGTGRCARRGAKSYACRRGCGGC
SAMN32324430   ACGGCGCCGCCTGTGACAGNGATNNANNNGCGGN
```

I can't see a single site with information. Every site I've looked at has either a base (A,C,T,G) or a matching ambiguous base (e.g. `R` matches `A`). In parsimony terms, these are strictly uninformative. In likelihood they are not quite as strictly uninformative, but likelihood sums over all possible states, and so the **vast** majority of the likelihood will come from the state which matches the non-ambiguous bases. In other words, it's not excatly the same as parsimony, but it's functionally *very* similar in this respect. That's why your tree has ~zero branch lengths.

Hope some of that helps,

Rob

**Reply by sprocha** [reply]
Hi!
thanks for your answer. I faced a similar question recently when checking the outputs of a snp dataset. iqtree)2) states me :
Alignment has 449 sequences with 407966 columns, 407963 distinct patterns
87722 parsimony-informative, 30609 singleton sites, 289635 constant sites

Am I correct to assume these 289635 "constant" sites are not constant in the sense of identical or monomorphic (these are nor supposed to exist here) but are collumns with, say C and Y  or A and R (and N), which are stated as "constant" here... ?

Thanks,
sara

**Reply by tylerbreech** [reply]
Thank you! that is helpful; judging by the differences between all the individuals already known, it must be a filtering error.

**Reply by bqminh** [reply]
@sprocha The definition of constant (invariant) sites is explained in this FAQ: http://www.iqtree.org/doc/Frequently-Asked-Questions#why-does-iq-tree-complain-about-the-use-of-asc-model

**Reply by sprocha** [reply]
thanks a lot @bqminh !

**Reply by roblanf** [reply]
If you guys are dealing with SNP datasets then you might want to convert the meaning of the genotype calls to something that phylogenetics software would understand. 

For example, if you have a site where a diploid individual 1 has genotype CC, and individual 2 has genotype CT, representing this as

```
ind1    C
ind2    Y
```

tells IQ-TREE (and other software) that you have a 'C' and a base you're not sure if it's a C or T. That's not what you want.

In your case, you know that it's actually:

```
ind1    CC
ind2    CT
```

And this is something that likelihood and bayesian software can interpret properly - one constant site and one where there's a difference.

Rob

**Reply by sprocha** [reply]
thanks @roblanf ! aware of the meaning(less) of the analyses... just kind of exploratory one.

**Reply by bqminh** [reply]
Perhaps you should apply a genotype model: GT16 for phased genotypes or GT10 for unphased genotypes. Have look at this paper: https://doi.org/10.1186/s13059-021-02583-w, these models are implemented in RAxML-NG.
